### ISSUE TITLE
DX12 support

### ISSUE THREAD

**Issue by noorus** [mleise/livesplit_overlay#3]
Hi!
Can I ask what it would take to support DX12 for this?  
Is it a potentially trivial change or some fundamental incompatibility?

**Comment by mleise** [comment]
I think it is neither trivial nor fundamentally incompatible. Originally I wrote this for Alien: Isolation (DX11) and didn't notice how broken it was on Vulkan and DX12 until I tried it on Callisto Protocol. For all but DX12 I could get away with just uploading the texture after a frame has been rendered, but DX12 really means business when it comes to multi-threading and synchronization of CPU/GPU commands. I don't feel the energy to deep dive into that.

It is likely trivial for someone who wrote a DX12 render engine before. But even then, since we are advised to use ReShade's elegant wrapper API that unifies everything from OpenGL to DX12, there might be gotchas with feature combinations that are not directly supported in ReShade or even bugs. In that case they could still use the DX12 API directly, or they could ask for a fix and wait for a ReShade update.

**Comment by noorus** [comment]
Thanks for the response! Got it.  
Aside from the fullscreen support, this is also a very useful tool for simple streaming to friends, but the DX12 limitation is holding it back quite a bit. I might end up looking into it myself, though I'm not familiar with either ReShade nor dx12.

**Comment by mleise** [comment]
Yeah, why not. If you have time and energy for a little side project, I'd welcome the contribution. You might even end up writing you own add-ons to show off to your friends. You could visualize what's going on in a single-player game as a mini-map overlay or something.
